blamo/lib/Test/test.dart

```diff
@@ -0,0 +1,94 @@
+import 'package:blamo/main.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_form_builder/flutter_form_builder.dart';
+import 'package:intl/intl.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+class TestPage extends StatefulWidget {
+  StateData pass; //Passes the StateData object to the stateful constructor
+
+  TestPage(this.pass);
+  @override
+  _TestPageState createState() => new _TestPageState(pass);
+}
+
+class _TestPageState extends State<TestPage> {
+  final routeName = '/TestPage';
+  StateData currentState;
+  _TestPageState(this.currentState);
+
+  @override
+  final GlobalKey<FormBuilderState> _fbKey = GlobalKey<FormBuilderState>();
+  Widget build(BuildContext context) {
+    if(currentState.currentRoute != null) {
+      currentState.currentRoute = '/TestPage'; //Assigns currentState.currentRoute to the name of the current named route
+    }
```

Comment:
I believe you need this if statement for both Log info and Unit. This statement makes the nav bar close instead of generating a new page of where you are just at

Reply:
Yeah he will need that for each page.

Reply:
updated